iothub/device/tests/Pipeline/ClientTransportStateMachineTests.cs

```diff
@@ -0,0 +1,138 @@
+        [DataRow(ClientTransportState.Open, ClientStateAction.CloseComplete, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Open, ClientStateAction.CloseComplete, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Open, ClientStateAction.ConnectionLost, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Open, ClientStateAction.ConnectionLost, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Open, ClientStateAction.ConnectionLost, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenStart, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenStart, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenStart, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenStart, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenSuccess, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenSuccess, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenSuccess, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenSuccess, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenFailure, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenFailure, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenFailure, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.OpenFailure, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseStart, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseStart, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseStart, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseStart, ClientTransportState.Closed)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseComplete, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseComplete, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.CloseComplete, ClientTransportState.Closing)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.ConnectionLost, ClientTransportState.Opening)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.ConnectionLost, ClientTransportState.Open)]
+        [DataRow(ClientTransportState.Closing, ClientStateAction.ConnectionLost, ClientTransportState.Closing)]
```

Comment:
what would be a valid state here?

[DataRow(ClientTransportState.Closing, ClientStateAction.ConnectionLost, ClientTransportState.Closing)]